### PR TITLE
[Incremental, Driver] Only emit a real make-style dependency file for one frontend job.

### PR DESCRIPTION
**Pull request by davidungar**
Because every frontend job emits the same make-style dependency file, the build system uses a lot of time to read them all, slowing incremental builds of projects with many files.

This changes adds a driver flag, `-only-one-dependency-file`, which causes the driver to omit the -emit-dependency-path argument for subsequent frontend jobs, instead creating dummy files.

Not tested yet, except on a dummy 3,000 file project where it reduced incremental compilation time from ~40 secs to ~3 secs.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test source compatibility

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test source compatibility

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please test

File: lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -951,7 +951,11 @@ Driver::buildCompilation(const ToolChain &TC,
         ArgList->hasArg(options::OPT_driver_time_compilation);
     std::unique_ptr<UnifiedStatsReporter> StatsReporter =
         createStatsReporter(ArgList.get(), Inputs, OI, DefaultTargetTriple);
+
+    const bool OnlyOneDependencyFile =
+        !ArgList->hasArg(options::OPT_disable_only_one_dependency_file);
```
Comment (CodaFi) [inline lib/Driver/Driver.cpp]: We probably want to rephrase this as a positive flag.

**Review by CodaFi — APPROVED** [review]
General approach LGTM.  This needs a test to ensure we're emitting empty make files.

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
Resolves the worst of the problems with rdar://57824038
